source/guide_openslides.rst

```diff
@@ -0,0 +1,208 @@
+.. highlight:: console
+
+.. author:: GodMod <godmod@eqdkp-plus.eu>
+.. tag:: lang-python
+.. tag:: django
+.. tag:: presentations
+.. tag:: web
+.. tag:: assemblies
+.. tag:: meetings
+.. tag:: audience-business
+
+.. sidebar:: About
+
+  .. image:: _static/images/openslides.svg
+      :align: center
+
+##########
+OpenSlides
+##########
+
+.. tag_list::
+
+OpenSlides is a free, web-based presentation and assembly system for managing and projecting agenda, motions, and elections of assemblies.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`Python <lang-python>` and its package manager pip
+  * :manual:`supervisord <daemons-supervisord>`
+  * :manual:`Web Backends <web-backends>`
+
+License
+=======
+
+OpenSlides_ is released under the `MIT License`_.
+
+Prerequisites
+=============
+
+Your URL needs to be setup:
+
+::
+
+ [isabell@stardust ~]$ uberspace web domain list
+ isabell.uber.space
+ [isabell@stardust ~]$
+
+Installation
+============
+
+Step 1
+------
+Install the OpenSlides Python package:
+
+::
+
+ [isabell@stardust ~]$ pip3.9 install openslides --user
+ [...]
+ Running setup.py install for pyrsistent ... done
+ Running setup.py install for PyPDF2 ... done
+ Running setup.py install for roman ... done
+ Running setup.py install for websockets ... done
+ Running setup.py install for openslides ... done
+ [...]
+ [isabell@stardust ~]$
+
+Check if OpenSlides is installed by typing:
+
+::
+
+ [isabell@stardust ~]$ openslides --version
+ 3.3
+ [isabell@stardust ~]$
+
+Step 2
+------
+
+.. note::
+
+    OpenSlides is running on port 8000.
+
+.. include:: includes/web-backend.rst
+
+Step 3
+------
+Start the OpenSlides binary:
+
+::
+
+ [isabell@stardust ~]$ openslides runserver 0.0.0.0:8000
+ [2021-04-03 18:35:10 +0200] [16175] [INFO] openslides.utils.schema_version [SweN] No old schema version
+ [...]
+ Django version 2.2.19, using settings 'settings'
+ Starting ASGI/Channels version 2.3.1 development server at http://0.0.0.0:8000/
```

Comment:
Using the development server in production is slow a possibly insecure. Something like gunicorn is probably better suited here. Have a look at the [pretix guide](https://lab.uberspace.de/guide_pretix.html#step-8) for a simple example.

Reply:
Using Python 3.9, I was not able to install the `httptools` package, which is a requirement for `uvicorn` (openslides is a ASGI application). 
```
 [ostest6@caelum~]$ pip3.9 install httptools
[...]
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/include/python3.9 -I/tmp/pip-install-47j7kxsb/httptools/vendor/http-parser -c httptools/parser/parser.c -o build/temp.linux-x86_64-3.9/httptools/parser/parser.o -O2
    httptools/parser/parser.c:20:10: fatal error: Python.h: No such file or directory
       20 | #include "Python.h"
          |          ^~~~~~~~~~
    compilation terminated.
    error: command '/opt/rh/devtoolset-9/root/usr/bin/gcc' failed with exit code 1
```
I also tried everything under Python 3.8, but this creates other errors.
Therefore, I cannot currently use gunicorn, but I use daphne, which is an ASGI server that django ships with it.